packages/core/src/utils/ethapi.ts

```diff
@@ -226,4 +234,14 @@ export class EthApi {
       throw new Error(CivilErrors.EvmException);
     }
   }
+
+  private setAccount = (newAccount: EthAddress): void => {
```

Comment:
To make this a little more generic you could use the events package.

```
class EthApi extends Events {
  public set account(): void {
      // ...whatever
      this.trigger("set:account", data);
  }
}
```
then you could bind to it at any time instead of passing it in at construction only.  Probably want to put the actual event on Civil.ts